settings.py.txt
"""
Django settings for administration project.

Generated by 'django-admin startproject' using Django 3.1.1.

For more information on this file, see
https://docs.djangoproject.com/en/3.1/topics/settings/

For the full list of settings and their values, see
https://docs.djangoproject.com/en/3.1/ref/settings/
"""
import os
import sys
from pathlib import Path

# Build paths inside the project like this: BASE_DIR / 'subdir'.
BASE_DIR = Path(__file__).resolve().parent.parent


# Quick-start development settings - unsuitable for production
# See https://docs.djangoproject.com/en/3.1/howto/deployment/checklist/

# SECURITY WARNING: keep the secret key used in production secret!
SECRET_KEY = '<SECRET_KET>'

# SECURITY WARNING: don't run with debug turned on in production!
DEBUG = True

ALLOWED_HOSTS = []


# Application definition

INSTALLED_APPS = [
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    'django.contrib.humanize',
    'app'
    ]

MIDDLEWARE = [
    'django.middleware.security.SecurityMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.common.CommonMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
]
ROOT_URLCONF = 'administration.urls'

TEMPLATES = [
    {
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'DIRS': [os.path.join(BASE_DIR, 'templates')],
        'APP_DIRS': True,
        'OPTIONS': {
            'context_processors': [
                'django.template.context_processors.debug',
                'django.template.context_processors.request',
                'django.contrib.auth.context_processors.auth',
                'django.contrib.messages.context_processors.messages',
                'django.template.context_processors.static'
            ],
        },
    },
]


WSGI_APPLICATION = 'administration.wsgi.application'


# Database
# https://docs.djangoproject.com/en/3.1/ref/settings/#databases

DATABASES = {
        'default': {
            'ENGINE': 'django.db.backends.mysql',
            'NAME': 'admin_master',
            'USER': "root",
            'PASSWORD': "root",
            'PORT': "3306",
            # 'ATOMIC_REQUESTS': True,
            'OPTIONS': {
                'charset': 'utf8mb4',
            },
        }
    }

# Internationalization
# https://docs.djangoproject.com/en/3.1/topics/i18n/

LANGUAGE_CODE = 'en-us'

TIME_ZONE = 'UTC'

USE_I18N = True

USE_L10N = True

USE_TZ = True


# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/3.1/howto/static-files/


# STATIC FILES
STATIC_URL = '/static/'
STATICFILES_DIRS = (os.path.join(BASE_DIR, 'static'), )
STATIC_ROOT = ''

# MEDIA FILES
MEDIA_URL = '/media/'
MEDIA_ROOT = os.path.join(BASE_DIR, 'media')


CSV_DELIMITER = "|"

FILE_HEADER = 'H_DocType|H_ControlNo|H_AcctNo|H_CBType|H_CBDate|H_CBNumber|H_ResubNo|H_ResubDesc|' \
              'H_SuppName|H_SuppIDType|H_SuppID|H_DistName|H_DistIDType|H_DistID|H_SubClaimAmt|' \
              'H_TotalCONCount|L_ContractNo|L_ContractStatus|L_ShipToIDType|L_ShipToID|L_ShipToName|' \
              'L_ShipToAddress|L_ShipToCity|L_ShipToState|L_ShipToZipCode|L_ShipToHIN|L_InvoiceNo|' \
              'L_InvoiceDate|L_InvoiceLineNo|L_InvoiceNote|L_ItemNDCNo|L_ItemUPCNo|L_ItemQty|' \
              'L_ItemUOM|L_ItemWAC|L_ItemContractPrice|L_ItemCreditAmt|L_ShipTo340BID|L_ShipToGLN\n'


# EMAIL CONFIG
EMAIL_ACTIVE = True
EMAIL_HOST = 'smtp.gmail.com'
EMAIL_PORT = 587
EMAIL_HOST_USER = 'appempowerrm@gmail.com'
EMAIL_HOST_PASSWORD = 'EmpowerRM2019'
EMAIL_USE_TLS = True

# SESSIONS (USER ACTIVITY)
SESSION_EXPIRE_AT_BROWSER_CLOSE = True              # Expire the session when the user closes their browser (Django)
TIMEOUT_TO_CLOSE_USER_SESSION = 1800                # timeout (in seconds) to close inactivity user session (30 mins)
TIMEOUT_TO_ALERT_USER_SESSION = 1500                # timeout (in seconds) to close inactivity user session (25 mins)
DELTA_TO_CHECK_USER_SESSION = 60000                 # delta (in miliseconds) to check user session activity  (each 1min)


# FOLDERS STRUCTURE
CLIENTS_DIRECTORY = os.path.join('D:\django_project\Empower', 'Clients')

# 844 Folders
DIR_NAME_844_ERM_INTAKE = '844_ERM_intake'
DIR_NAME_844_PROCESSED = '844_processed'
DIR_NAME_844_ERM_ERROR = '844_errors'
DIR_NAME_844_ERM_HISTORY = '844_ERM_history'

# 849 Folders
DIR_NAME_849_ERM_OUT = '849_ERM_out'
DIR_NAME_849_ERM_HISTORY = '849_history'
DIR_NAME_849_ERM_MANUAL = '849_manual_files'
# User Files Storage
DIR_NAME_FILES_STORAGE = 'files_storage'
DIR_NAME_USER_REPORTS = 'reports'
DIR_NAME_USER_REPORTS_SCHEDULE = 'reports_schedule'


FOLDERS_STRUCTURE = {
    DIR_NAME_844_ERM_INTAKE: "844s Waiting for Import",
    DIR_NAME_844_PROCESSED: '844_processed',
    DIR_NAME_844_ERM_ERROR: "844s Unable to be processed",
    DIR_NAME_849_ERM_OUT: "849s Waiting for Pickup",
    DIR_NAME_FILES_STORAGE: "User Uploads",
    DIR_NAME_849_ERM_HISTORY: "849s History",
    DIR_NAME_849_ERM_MANUAL: "849 Excel Files",
    DIR_NAME_USER_REPORTS: "Reports",
    DIR_NAME_USER_REPORTS_SCHEDULE: 'Reports Schedule'
}




os.environ["DJANGO_ALLOW_ASYNC_UNSAFE"] = "true"
EDI_API_TOKEN = "180E843A545143D384C24D1135D5D567"
BASE_API_URL = "http://localhost:8003/api"


Hello Chetan.


administration/settings.py
"""
Django settings for administration project.

Generated by 'django-admin startproject' using Django 3.1.1.

For more information on this file, see
https://docs.djangoproject.com/en/3.1/topics/settings/

For the full list of settings and their values, see
https://docs.djangoproject.com/en/3.1/ref/settings/
"""
import os
import sys
from pathlib import Path

# Build paths inside the project like this: BASE_DIR / 'subdir'.
BASE_DIR = Path(__file__).resolve().parent.parent


# Quick-start development settings - unsuitable for production
# See https://docs.djangoproject.com/en/3.1/howto/deployment/checklist/

# SECURITY WARNING: keep the secret key used in production secret!
SECRET_KEY = '<SECRET_KET>'

# SECURITY WARNING: don't run with debug turned on in production!
DEBUG = True

ALLOWED_HOSTS = ['*']


# Application definition

INSTALLED_APPS = [
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    'django.contrib.humanize',
    'app'
    ]

MIDDLEWARE = [
    'django.middleware.security.SecurityMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.common.CommonMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
]
ROOT_URLCONF = 'administration.urls'

TEMPLATES = [
    {
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'DIRS': [os.path.join(BASE_DIR, 'templates')],
        'APP_DIRS': True,
        'OPTIONS': {
            'context_processors': [
                'django.template.context_processors.debug',
                'django.template.context_processors.request',
                'django.contrib.auth.context_processors.auth',
                'django.contrib.messages.context_processors.messages',
                'django.template.context_processors.static'
            ],
        },
    },
]


WSGI_APPLICATION = 'administration.wsgi.application'


# Database
# https://docs.djangoproject.com/en/3.1/ref/settings/#databases

DATABASES = {
        'default': {
            'ENGINE': 'django.db.backends.mysql',
            'NAME': 'admin_master',
            'USER': "root",
            'PASSWORD': "8cWV@}nd",
            'PORT': "3306",
            # 'ATOMIC_REQUESTS': True,
            'OPTIONS': {
                'charset': 'utf8mb4',
            },
        }
    }

# Internationalization
# https://docs.djangoproject.com/en/3.1/topics/i18n/

LANGUAGE_CODE = 'en-us'

TIME_ZONE = 'UTC'

USE_I18N = True

USE_L10N = True

USE_TZ = True


# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/3.1/howto/static-files/


# STATIC FILES
STATIC_URL = '/static/'
#STATICFILES_DIRS = (os.path.join(BASE_DIR, 'static'), )
STATIC_ROOT = os.path.join(BASE_DIR, 'static')
#STATIC_ROOT = ''

# MEDIA FILES
MEDIA_URL = '/media/'
MEDIA_ROOT = os.path.join(BASE_DIR, 'media')


CSV_DELIMITER = "|"

FILE_HEADER = 'H_DocType|H_ControlNo|H_AcctNo|H_CBType|H_CBDate|H_CBNumber|H_ResubNo|H_ResubDesc|' \
              'H_SuppName|H_SuppIDType|H_SuppID|H_DistName|H_DistIDType|H_DistID|H_SubClaimAmt|' \
              'H_TotalCONCount|L_ContractNo|L_ContractStatus|L_ShipToIDType|L_ShipToID|L_ShipToName|' \
              'L_ShipToAddress|L_ShipToCity|L_ShipToState|L_ShipToZipCode|L_ShipToHIN|L_InvoiceNo|' \
              'L_InvoiceDate|L_InvoiceLineNo|L_InvoiceNote|L_ItemNDCNo|L_ItemUPCNo|L_ItemQty|' \
              'L_ItemUOM|L_ItemWAC|L_ItemContractPrice|L_ItemCreditAmt|L_ShipTo340BID|L_ShipToGLN\n'


# EMAIL CONFIG
EMAIL_ACTIVE = True
EMAIL_HOST = 'smtp.gmail.com'
EMAIL_PORT = 587
EMAIL_HOST_USER = 'appempowerrm@gmail.com'
EMAIL_HOST_PASSWORD = 'EmpowerRM2019'
EMAIL_USE_TLS = True

# SESSIONS (USER ACTIVITY)
SESSION_EXPIRE_AT_BROWSER_CLOSE = True              # Expire the session when the user closes their browser (Django)
TIMEOUT_TO_CLOSE_USER_SESSION = 1800                # timeout (in seconds) to close inactivity user session (30 mins)
TIMEOUT_TO_ALERT_USER_SESSION = 1500                # timeout (in seconds) to close inactivity user session (25 mins)
DELTA_TO_CHECK_USER_SESSION = 60000                 # delta (in miliseconds) to check user session activity  (each 1min)


# FOLDERS STRUCTURE
CLIENTS_DIRECTORY = "".join(os.path.join('/', 'efs/baseline/'))

# 844 Folders
DIR_NAME_844_ERM_INTAKE = '844_ERM_intake'
DIR_NAME_844_PROCESSED = '844_processed'
DIR_NAME_844_ERM_ERROR = '844_errors'
DIR_NAME_844_ERM_HISTORY = '844_ERM_history'

# 849 Folders
DIR_NAME_849_ERM_OUT = '849_ERM_out'
DIR_NAME_849_ERM_HISTORY = '849_history'
DIR_NAME_849_ERM_MANUAL = '849_manual_files'
# User Files Storage
DIR_NAME_FILES_STORAGE = 'files_storage'
DIR_NAME_USER_REPORTS = 'reports'
DIR_NAME_USER_REPORTS_SCHEDULE = 'reports_schedule'


FOLDERS_STRUCTURE = {
    DIR_NAME_844_ERM_INTAKE: "844s Waiting for Import",
    DIR_NAME_844_PROCESSED: '844_processed',
    DIR_NAME_844_ERM_ERROR: "844s Unable to be processed",
    DIR_NAME_849_ERM_OUT: "849s Waiting for Pickup",
    DIR_NAME_FILES_STORAGE: "User Uploads",
    DIR_NAME_849_ERM_HISTORY: "849s History",
    DIR_NAME_849_ERM_MANUAL: "849 Excel Files",
    DIR_NAME_USER_REPORTS: "Reports",
    DIR_NAME_USER_REPORTS_SCHEDULE: 'Reports Schedule'
}




os.environ["DJANGO_ALLOW_ASYNC_UNSAFE"] = "true"
EDI_API_TOKEN = "180E843A545143D384C24D1135D5D567"
BASE_API_URL = "http://172.31.71.125:8025/api"
